safe_transaction_service/history/services/notification_service.py

```diff
@@ -204,3 +205,54 @@ def build_reorg_payload(block_number: int) -> ReorgPayload:
         blockNumber=block_number,
         chainId=str(get_chain_id()),
     )
+
+
+class DelegatePayload(TypedDict):
+    type: str
+    safeAddress: Optional[str]
+    delegate: str
+    delegator: str
+    label: str
+    expiryDate: Optional[int]
+
+
+def _build_delegate_payload(
+    event_type: Union[
+        TransactionServiceEventType.NEW_DELEGATE,
+        TransactionServiceEventType.UPDATED_DELEGATE,
+        TransactionServiceEventType.DELETED_DELEGATE,
+    ],
+    instance: SafeContractDelegate,
+) -> DelegatePayload:
+    """
+    Build a delegate payload with the specified event type and SafeContractDelegate instance data.
+
+    :param event_type: The transaction event type, restricted to NEW_DELEGATE, UPDATED_DELEGATE, or DELETED_DELEGATE.
+    :param instance: An instance of SafeContractDelegate.
+    :return: A DelegatePayload dictionary with details about the delegate.
+    """
+    return DelegatePayload(
+        type=event_type.name,
+        safeAddress=instance.safe_contract.address if instance.safe_contract else None,
+        delegate=instance.delegate,
+        delegator=instance.delegator,
+        label=instance.label,
+        expiryDate=(
+            int(instance.expiry_date.timestamp()) if instance.expiry_date else None
+        ),
+    )
+
+
+def build_save_delegate_payload(
+    instance: SafeContractDelegate, created: bool = True
+) -> DelegatePayload:
+    event_type = TransactionServiceEventType.NEW_DELEGATE
+    if not created:
+        event_type = TransactionServiceEventType.UPDATED_DELEGATE
```

Comment:
Seems more clear this way

```suggestion
    if created:
        event_type = TransactionServiceEventType.NEW_DELEGATE
    else:
        event_type = TransactionServiceEventType.UPDATED_DELEGATE
```